File: app/src/main/java/com/rr/rhythms/Interfaces/IGraph.java
package com.rr.rhythms.Interfaces;

import com.rr.rhythms.Entities.GraphInnerState;
import com.rr.rhythms.Entities.Point;

import java.util.ArrayList;

/**
 * Created by Huruk on 7/5/2016.
 */
public interface IGraph {
    ArrayList<Point> getPoints();

    int getColor();

    String getName();

    void generateGraph(int totalDays, int scale);

    GraphInnerState[] getCurrentStates(int todalDays);
}

File: app/src/main/java/com/rr/rhythms/Interfaces/IGraph.kt
package com.rr.rhythms.Interfaces

import com.rr.rhythms.Entities.GraphInnerState
import com.rr.rhythms.Entities.Point

import java.util.ArrayList

/**
 * Created by Huruk on 7/5/2016.
 */
interface IGraph {
    val points: ArrayList<Point>

    val name: String

    val color: Int

    fun generateGraph(totalDays: Int, scale: Int)

    fun getCurrentStates(todalDays: Int): Array<GraphInnerState>
}
